src/Microsoft.ML.LightGBM/LightGbmRankingTrainer.cs

```diff
@@ -0,0 +1,145 @@
+        }
+
+        public override PredictionKind PredictionKind { get { return PredictionKind.Ranking; } }
+    }
+
+    public sealed class LightGbmRankingTrainer : LightGbmTrainerBase<float, LightGbmRankingPredictor>
+    {
+        public const string Summary = "LightGBM Ranking";
+        public const string LoadNameValue = "LightGBMRanking";
+        public const string ShortName = "LightGBMRank";
+
+        public LightGbmRankingTrainer(IHostEnvironment env, LightGbmArguments args)
+            : base(env, args, PredictionKind.Ranking, "LightGBMRanking")
+        {
+        }
+
+        protected override void CheckDataValid(IChannel ch, RoleMappedData data)
+        {
+            Host.AssertValue(ch);
+            base.CheckDataValid(ch, data);
+            // Check label types.
+            var labelType = data.Schema.Label.Type;
+            if (!(labelType.IsKey || labelType == NumberType.R4))
+            {
+                throw ch.ExceptParam(nameof(data),
+                    $"Label column '{data.Schema.Label.Name}' is of type '{labelType}', but must be key or R4.");
+            }
+            // Check group types.
+            var groupType = data.Schema.Group.Type;
+            if (!(groupType == NumberType.U4 || groupType.IsKey))
+            {
+                throw ch.ExceptParam(nameof(data),
+                   $"Group column '{data.Schema.Group.Name}' is of type '{groupType}', but must be U4 or a Key.");
+            }
+        }
+
+        public override LightGbmRankingPredictor CreatePredictor()
+        {
+            Host.Check(TrainedEnsemble != null, "The predictor cannot be created before training is complete");
+            var innerArgs = LightGbmInterfaceUtils.JoinParameters(Options);
+            return new LightGbmRankingPredictor(Host, TrainedEnsemble, FeatureCount, innerArgs);
+        }
+
+        protected override void CheckAndUpdateParametersBeforeTraining(IChannel ch, RoleMappedData data, float[] labels, int[] groups)
+        {
+            Host.AssertValue(ch);
+            Options["objective"] = "lambdarank";
+            ch.CheckValue(groups, nameof(groups));
+            // Add default metric.
+            if (!Options.ContainsKey("metric"))
+                Options["metric"] = "ndcg";
+            // Only output one ndcg score.
+            Options["eval_at"] = "5";
+        }
+    }
+
+    /// <summary>
+    /// A component to train an LightGBM model.
+    /// </summary>
+    public static partial class LightGbm
+    {
+        [TlcModule.EntryPoint(Name = "Trainers.LightGbmRanker", Desc = "Train an LightGBM ranking model", UserName = LightGbmRankingTrainer.Summary, ShortName = LightGbmRankingTrainer.ShortName)]
+        public static CommonOutputs.RankingOutput TrainRanking(IHostEnvironment env, LightGbmArguments input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            var host = env.Register("TrainLightGBM");
```

Comment:
>TrainLightGBM [](start = 37, length = 13)

would be nice to have specific to kind channel, other than same string across all 4 trainers.